cpp/include/cudf/detail/utilities/transform_unary_functions.cuh

```diff
@@ -30,6 +30,21 @@
 #include <thrust/pair.h>
 
 namespace cudf {
+
+/**
+ * @brief Functor to static_cast convertible types.
+ *
+ * @tparam result_type Target type to be typecasted to.
+ */
+template <typename ResultType>
+struct typecaster {
```

Comment:
this assumes that `ElementType` is convertible to `ResultType` which may not be case always.

Reply:
That will create a compilation error. not a runtime error.